integration_test/list_test.go

```diff
@@ -42,3 +45,122 @@ func TestKrewList(t *testing.T) {
 	}
 	// TODO(ahmetb): install multiple plugins and see if the output is sorted
 }
+
+func TestKrewListJSON(t *testing.T) {
+	skipShort(t)
+
+	test, cleanup := NewTest(t)
+	defer cleanup()
+
+	test.Krew("install",
+		"--manifest", filepath.Join("testdata", "foo.yaml"),
+		"--archive", filepath.Join("testdata", "foo.tar.gz")).
+		RunOrFail()
+
+	expected := []byte("{\n    \"Name\": \"foo\",\n    \"Version\": \"v0.1.0\"\n}\n")
+
+	eventualList := test.WithIndex().Krew("list", "-o", "json").RunOrFailOutput()
+	if diff := cmp.Diff(eventualList, expected); diff != "" {
+		t.Fatalf("'list' output doesn't match:\n%s", diff)
+	}
+
+	Plugin := struct {
+		Name    string
+		Version string
+	}{"", ""}
+
+	err := json.Unmarshal(eventualList, &Plugin)
+	if err != nil || Plugin.Name != "foo" || Plugin.Version != "v0.1.0" {
+		t.Fatalf("Error unmarshaling: %s. Plugin: \"%s\". Version: \"%s\".", err, Plugin.Name, Plugin.Version)
+	}
+}
+
+func TestKrewListJSONEmpty(t *testing.T) {
+	skipShort(t)
+
+	test, cleanup := NewTest(t)
+	defer cleanup()
+
+	initialList := test.WithIndex().Krew("list", "-o", "json").RunOrFailOutput()
+	initialOut := []byte{}
+
+	if diff := cmp.Diff(initialList, initialOut); diff != "" {
+		t.Fatalf("expected empty output from 'list':\n%s", diff)
+	}
+}
+
+func TestKrewListYAML(t *testing.T) {
+	skipShort(t)
+
+	test, cleanup := NewTest(t)
+	defer cleanup()
+
+	test.Krew("install",
+		"--manifest", filepath.Join("testdata", "foo.yaml"),
+		"--archive", filepath.Join("testdata", "foo.tar.gz")).
```

Comment:
fwiw if you install 2 plugins you can test the ordering (sorted by name)

Reply:
Implemented sorting tests for all output types.